keps/sig-api-machinery/2161-apiserver-default-labels/README.md

```diff
@@ -0,0 +1,452 @@
+
+### Version Skew Strategy
+
+It is assumed that losing data under a `kubernetes.io` label is acceptable, since its a reserved label and should not generally be used by non-k8s core components:
+
+```
+The kubernetes.io/ and k8s.io/ prefixes are reserved for Kubernetes core components.
+```
+
+which is published in https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/.
+
+Since this has been reserved since 2017 (if not earlier), no backwards guarantees are made for clusters using the `kubernetes.io/metadata.name` label, but we'll do our best to not break people who coincidentally may have used this label (debatable wether this is a realistic concern or not).
+
+- Old versions will be assumed to never have this label.  If they do have this label, then they will eventually lose its contents.
+- New versions always will have this label, lazily added on read or write.
+- During upgrades, a policy created may be able to "get away" with temporarily writing to the aformentioned "kubernetes.io" metadata.name label, however, this write will eventually be lost.
+- It's generally agreed that the kubernetes.io prefix is reserved, and thus, there is no need to warn or carefully migrate its presence into releases. We will thus simply have a feature gate, which can be disabled for one release.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+1.21:
+- enable by default
+- allow gate to disable the feature
+- release note
+
+1.22:
+- gate enabled by default
+- remove gate
+- release note
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  Yes, network policies which are already using this label, may break, although such policies would be in violation of the standard that k8s.io labels are reserved.
+
+* **What specific metrics should inform a rollback?**
+  No e2e tests or metrics exist to inform this, but if a net loss of healthy pods occurred and a user is actively using networkpolicies that impersonate on the k8s.io label, that would be a hint. 
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+  Not yet. TBD.
+  
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs, 
+  Any user relying on namespaces with **no** labels, or expecting **some** namespaces to not have labels including the `k8s.io`prefix defined here, would observe a potential changes in namespace selection behaviour with this, however, that is not a guaranteed API feature in any way, so no actual feature is being deprecated. 
+  
+### Monitoring Requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can an operator determine if the feature is in use by workloads?**
+
+By running `kubectl get ns --show-labels` and inspecting the `kubernetes.io` values.
```

Comment:
this isn't quite right.  You'd need to track list/watch requests keying against this name.